test.js

```diff
@@ -76,6 +76,14 @@ test('should convert multiaddr to URI', (t) => {
     [
       '/ip4/1.2.3.4/tcp/3456/p2p/QmcNwyju7SWoizsAuf6kjaaRoxe762ovsT3hz6qt3xxcsK',
       'tcp://1.2.3.4:3456/p2p/QmcNwyju7SWoizsAuf6kjaaRoxe762ovsT3hz6qt3xxcsK'
+    ],
+    [
+      '/onion3/pzbmpqsyi2j2za2fwmiognupwozjlcoajbd3cgboetql7ooah63zywqd:1111',
```

Comment:
Implicit TCP is bit scary here.  I know `HiddenServicePort` does not support UDP _right now_, but it might one day, and using this implicit notation feels risky. 

Tor Browser and Brave support `.onion` TLD defined in  [RFC7686:  The ".onion" Special-Use Domain Name](https://tools.ietf.org/html/rfc7686) and regular `/dns4/pzbmpqsyi2j2za2fwmiognupwozjlcoajbd3cgboetql7ooah63zywqd.onion/tcp/1111/`  should work fine, as long your DNS resolver performs `.onion` lookup via Tor (eg. using `DNSPort` from `torrc`).

Given that, do we need `/onion3` if the `/dns4` works? If so, perhaps a better idea is to follow multiaddr convention and explicitly state that `1111`  is tcp?

```
/onion3/pzbmpqsyi2j2za2fwmiognupwozjlcoajbd3cgboetql7ooah63zywqd/tcp/1111/
```